Formula/asdf.rb

```diff
@@ -19,19 +20,35 @@ class Asdf < Formula
   depends_on "readline"
   depends_on "unixodbc"
 
-  conflicts_with "homeshick",
-    because: "asdf and homeshick both install files in lib/commands"
-
   def install
     bash_completion.install "completions/asdf.bash"
     fish_completion.install "completions/asdf.fish"
     zsh_completion.install "completions/_asdf"
-    libexec.install "bin/private"
-    prefix.install Dir["*"]
-    touch prefix/"asdf_updates_disabled"
+    libexec.install Dir["*"]
+    touch libexec/"asdf_updates_disabled"
+    bin.write_exec_script libexec/"bin/asdf"
```

Comment:
Shouldn't we consider this a placeholder file too (to be included below)?

Reply:
I think we want users to always be able to call `asdf` even if they don't source the setup scripts (even if this means limited functionality), so we'll still want to keep this even when we remove the rest.

Am I mistaken here? If so, we'll probably want to mention in the `caveats` that the additional setup is now mandatory for anything to work at all.

Reply:
@carlocab, actually the additional setup was always mandatory, and still is. Having just asdf on the path, and being able to run it and see the help message output, might give users the false impression that they already have it set up, but the setup script is required. For example, the script adds the shims directory to the path, which is absolutely necessary for asdf to work.

Reply:
Still, some things work without them. @jthegedus, how common is it for users to find themselves misled by having `asdf` in their `PATH` without having done the proper setup?

Reply:
I can believe that the lion's share of those users installed asdf using Homebrew -- that doesn't tell me how often that happens though.

On the other hand, I feel like we might be trading one issue for another here. That is, we might get fewer users who never sourced the set up scripts and are confused by asdf not working, but we also might get more users who did `brew install asdf` and are confused about why calling `asdf` returns `command not found`.

Not sure which one's better -- might work out to be effectively the same either way.

Reply:
Not sure I get which are the two issues we're trading here. The "users who never sourced the set up scripts" never had and won't have asdf working as expected. If they try to call asdf and get a "command not found" error, it's just a different symptom that they didn't source the set up scripts yet, and don't have asdf working as expected either due to that. We're just letting it 
"fail fast" now, which I would argue is better!  :- )  Also, we're adding the caveats, in the hope users will read that -- if they didn't read any asdf docs/repo etc., because they "just run brew install asdf without any research", at least they have the caveats, which is typically what Homebrew users look at.

Reply:
I do not have concrete numbers. I agree with @ericbn here and is why, in many a prior PR, I was trying to get caveats instituted to alert users.

Reply:
Ok, fair enough, if there truly cannot be a user with a use-case for doing `brew install asdf` then calling `asdf` for some reason without sourcing the setup scripts, we can move this line back down.